crypto/params.c

```diff
@@ -1414,32 +1442,22 @@ static int set_string_internal(OSSL_PARAM *p, const void *val, size_t len,
 
 int OSSL_PARAM_set_utf8_string(OSSL_PARAM *p, const char *val)
 {
-    if (p == NULL) {
+    if (p == NULL || val == NULL) {
         err_null_argument;
         return 0;
     }
-
     p->return_size = 0;
-    if (val == NULL) {
-        err_null_argument;
-        return 0;
-    }
     return set_string_internal(p, val, strlen(val), OSSL_PARAM_UTF8_STRING);
 }
 
 int OSSL_PARAM_set_octet_string(OSSL_PARAM *p, const void *val,
                                 size_t len)
 {
-    if (p == NULL) {
+    if (p == NULL || val == NULL) {
         err_null_argument;
         return 0;
     }
-
     p->return_size = 0;
-    if (val == NULL) {
-        err_null_argument;
-        return 0;
-    }
```

Comment:
!

Reply:
Because I was moving the statement that separated the two tests out of the way, and they raise the same error, I folded the tests into one test. Seemed to make sense.